datasets/stream_spec.py

```diff
@@ -0,0 +1,192 @@
+            assert n_drifts == None and sequential_drifts == False, 'you must use n_slots, n_drifts or sequential_drifts arguments (cant use them together)'
+        if n_drifts is not None:
+            assert n_slots == None and sequential_drifts == False, 'you must use n_slots, n_drifts or sequential_drifts arguments (cant use them together)'
+        if sequential_drifts:
+            assert n_slots == None and n_drifts is None, 'you must use n_slots, n_drifts or sequential_drifts arguments (cant use them together)'
+
+        self.n_tasks = n_tasks
+        self.n_slots = n_slots
+        self.n_classes = n_classes
+        self.random_seed = random_seed
+        self.n_drifts = n_drifts
+        self.sequential_drifts = sequential_drifts
+
+        if n_drifts is not None:
+            self.task_classes = self.create_n_drifts()
+        elif sequential_drifts:
+            self.task_classes = self.create_sequential_drifts()
+        else:
+            self.task_classes = self.random_class_asigment()
+
+        self.current_task = 0
+
+    def create_n_drifts(self):
+        assert self.n_drifts < self.n_tasks
+
+        drift_duration = self.n_tasks // (self.n_drifts + 1)
+        drift_indexes = list()
+        for i in range(drift_duration, self.n_tasks, drift_duration):
+            if len(drift_indexes) == self.n_drifts:
+                break
+            drift_indexes.append(i)
+
+        if len(drift_indexes) < self.n_drifts:
+            drift_indexes.append(self.n_tasks-1)
+        print('Creating drifts at tasks: ', drift_indexes)
+
+        classes_per_task = self.n_classes // self.n_tasks
+        last_class = 0
+        last_drifted_class = -1
+        task_classes = list()
+        for t in range(self.n_tasks):
+            new_classes = list(range(last_class, last_class + classes_per_task))
+            if t in drift_indexes:
+                min_label = min(new_classes)
+                for c in range(last_drifted_class+1, min_label):
+                    new_classes.append(c)
+                last_drifted_class = min_label - 1
+            task_classes.append(sorted(new_classes))
+            last_class += classes_per_task
+
+        return task_classes
+
+    def create_sequential_drifts(self):
+        """each task form 1 to n-1 contains set of new classes and drifted classes from previous task"""
+        task_classes = list()
+
+        classes_per_task = self.n_classes // self.n_tasks
+        last_new_classes = []
+        assert classes_per_task >= 2
+        last_class = 0
+        for t in range(self.n_tasks):
+            new_classes = list(range(last_class, last_class + classes_per_task))
+            classes_to_add = list()
+            classes_to_add.extend(new_classes)
+            classes_to_add.extend(last_new_classes)
+            task_classes.append(sorted(classes_to_add))
+            last_class += classes_per_task
+            last_new_classes = copy.deepcopy(new_classes)
+        return task_classes
+
```

Comment:
This should be more a straightforward implementation for sequential drift:

`def create_sequential_drifts(self):
    assert self.n_classes % self.n_tasks == 0, 'Total number of classes should be divisible by number of tasks'
    classes_per_task = self.n_classes // self.n_tasks
    assert classes_per_task >= 2, 'At least two classes should be present in each new task'

    task_classes = []
    for t in range(self.n_tasks):
        new_classes = list(range(classes_per_task * t, classes_per_task * t + classes_per_task))
        old_classes = list(range(max(classes_per_task * t - classes_per_task, 0), classes_per_task * t))
        task_classes.append(old_classes + new_classes)

    return task_classes`

Reply:
You're right. This version is simpler. I will change it.
The only thing I'm not sure is the first assertion. There may be some benchmarks with some weird number of classes, where this would not be true, yet we could still use them.